napari_animation/animation.py

```diff
@@ -1,334 +1,171 @@
 
-        if type(self.inter_steps) is not list:
-            inter_steps = len(self.key_frames) * [self.inter_steps]
-        else:
-            inter_steps = self.inter_steps
-
-        empty = {
-            "ndisplay": [],
-            "frame": [],
-            "camera": [],
-            "vis": [],
-            "sliders": [],
-        }
-        states_dict = []
-        for ind, x in enumerate(self.key_frames):
-            states_dict.append(x)
-            # do not add frames after last key-frame
-            if ind < len(self.key_frames) - 1:
-                # do not add frames when switching camera
-                if (
-                    self.key_frames[ind]['ndisplay']
-                    == self.key_frames[ind + 1]['ndisplay']
-                ):
-                    for y in range(inter_steps[ind]):
-                        states_dict.append(copy.deepcopy(empty))
-        for ind, x in enumerate(states_dict):
-            x["frame"] = ind
-        self.states_dict = states_dict
-
-    def create_steps(self):
-        """Interpolate states between key-frames"""
-
-        self.create_state_dict()
-        self.interpolated_states = util.interpolate(self.states_dict)
-
-    def key_interpolframe(self, viewer):
-        """Progress through interpolated frames"""
-
-        self.create_steps()
-
-        new_frame = (self.current_interpolframe + 1) % len(self.states_dict)
-        self.update_viewer_from_state(new_frame)
-        self.current_interpolframe = new_frame
-
-    def update_viewer_from_state(self, frame):
-        """Set the viewer to a given interpolated frame
+    def _get_viewer_state(self):
+        """Capture current viewer state
 
-        Parameters
+        Returns
         -------
-        frame : int
-            frame to visualize
+        new_state : dict
+            Description of viewer state.
         """
 
-        # set view type 2D/3D and camera state
-        self.viewer.dims.ndisplay = self.interpolated_states["ndisplay"][frame]
-        self.viewer.window.qt_viewer.view.camera.set_state(
-            self.interpolated_states["camera"][frame]
-        )
-
-        # assign interpolated visibility state
-        for j in range(len(self.viewer.layers)):
-            self.viewer.layers[j].visible = self.interpolated_states["vis"][
-                frame
-            ][j]
-
-        # adjust slider positions
-        for i in range(self.interpolated_states["sliders"].shape[1]):
-            self.viewer.dims.set_point(
-                i, self.interpolated_states["sliders"][frame][i]
-            )
+        new_state = {
+            'camera': self.viewer.camera.asdict(),
+            'dims': self.viewer.dims.asdict(),
+        }
 
-        # update view
-        self.viewer.window.qt_viewer.view.camera.view_changed()
+        # Log transform zoom for linear interpolation
+        new_state['camera']['zoom'] = np.log10(new_state['camera']['zoom'])
```

Comment:
I don't love that i have to do this, I wonder if we want to store the log transformed zoom on the camera instead, or if there is a cleaner way to do this ... can figure out later though